Add example script documenting diagnostic features to docs (#1227)
pace-neutrons/Horace: documentation/user_docs/docs/example_scripts/Horace_diagnostics.rst

Horace Diagnostics
##################

This is a tutorial to explore some of the diagnostic features in
Horace that allow you to understand strange features in your data.

This will take you through all the steps from generation, spurion
spotting, and analysis.

Generating data
===============

::

   %Take a cut from an sqw file, to get a dispersion plot. Ensure we do not

>    %select the -nopix option, i.e. the object "cs" will be larger in memory,

Reviewer: ```suggestion
   %select the -nopix option. While this means that the object "cs" will be larger in memory than with `-nopix`,
```

>    %but will retain all of the information about detector pixels from

Reviewer: ```suggestion
   %it will retain all of the information about detector pixels from
```

   plot(compact(cs));

   lz 0 100;
   keep_figure;

::

   Taking cut from data in file /mnt/data/Science/URu2Si2/data/sqw/Ei81_20K.sqw...
   Step 1 of 1; Have read data for 488866 pixels -- now processing data... -----> retained 24161 pixels

This is not very exciting, but notice there is an intense streak at
K=-3, E=8. Is this real, or a spurion? One way to find out is to use
the run inspector tool. This plots out the contributions to a cut from
each of the individual runs (so long as you retained the detector
pixel info, as we have here).

Run inspector
=============

::

   %Invoke the run inspector:
   run_inspector(cs,'ax',[-4,-2,0,20],'col',[0,100]);

.. note::

   The option ``ax`` sets the limits of the plot as follows: [xmin,
   xmax, ymin, ymax].

   If you don't do this, then data from each run will be plotted with
   different axes, which makes them quite hard to compare.

.. note::

   The option ``col`` sets the colourmap scale [col_min,col_max]. Again,
   not setting this means data from each run will be plotted with
   different colour scale, which is not so useful.

.. note::

   You can explore the data, toggling between runs using the arrow keys,

>    and the page-up / page-down keys to skip 10 runs ahead.

Reviewer: ```suggestion
   and using the page-up / page-down keys to skip 10 runs ahead.
```

Notice that the value of psi and the run number from when you made the
data file are given in the plot title.

You should find that run 14 (psi=26) has a high point in the region
we're worried about. But runs 89 and 90 (psi=25 and 26), 150
(psi=26.5) and 209 (psi=25.5) do not have this feature.

This suggests that it is a spurion. We can test this by masking the
data just from this run in our object:

.. note::

   the 14 here corresponds to the 14th run in our complete dataset

::

   cs_m = mask_runs(cs,14);
   plot(compact(cs_m));

   keep_figure

and this confirms that indeed it was a spurion.

Understanding data from a single orientation
============================================

If you have a material which is strongly 1d or 2d you may well only
measure with the sample in a single orientation. You can then
integrate between +/- infinity on the axes where there is no
dispersion.  This improves statistics without the need to count lots
of different orientations and hence is much faster.

Generating data
---------------

Generate a suitable sqw file, from a single orientation (see Taylor et

> al) for an explanation of the science of this particular material.

Reviewer: ```suggestion
al for an explanation of the science of this particular material).
```
